### PR TITLE
Updated mutect2.wdl and funcotator.wdl with latest args for funcotator.

### LINKED ISSUE #4587: Funcotator - Need to update M2 WDL to allow for all options to be passed in for Funcotator
Need to review and fix the WDL for M2 to include the flags for hg38 and all other input options.

### PR DESCRIPTION
**Pull request by jonn-smith**
Fixes #4587

### CODE REVIEW

File: scripts/mutect2_wdl/mutect2.wdl [hunk 1]
```diff
@@ -350,22 +350,33 @@ workflow Mutect2 {
...
+
+                gatk_override                                       = gatk_override,
+
+                data_sources_tar_gz                                 = data_sources_tar_gz,
+                transcript_selection_mode                           = transcript_selection_mode,
+                transcript_selection_list                           = transcript_selection_list,
+                annotation_defaults                                 = annotation_defaults,
+                annotation_overrides                                = annotation_overrides,
+                filter_funcotations                                 = filter_funcotations_or_default,
+#                interval_list
```
Comment (davidbenjamin) [inline scripts/mutect2_wdl/mutect2.wdl]: Delete?

File: scripts/mutect2_wdl/mutect2.wdl [hunk 2]
```diff
@@ -350,22 +350,33 @@ workflow Mutect2 {
...
+                gatk_docker                                         = gatk_docker,
+
+                ref_fasta                                           = ref_fasta,
+                ref_fasta_index                                     = ref_fai,
+                ref_dict                                            = ref_dict,
+                input_vcf                                           = funcotate_vcf_input,
+                input_vcf_idx                                       = funcotate_vcf_input_index,
+                reference_version                                   = select_first([reference_version, "NO_REFERENCE_VERSION_GIVEN"]),
+                output_file_base_name                               = funcotated_name,
+                output_format                                       = "VCF",
```
Comment (davidbenjamin) [inline scripts/mutect2_wdl/mutect2.wdl]: Just checking that hard-coding VCF is intentional.

File: scripts/mutect2_wdl/mutect2.wdl [hunk 3]
```diff
@@ -350,22 +350,33 @@ workflow Mutect2 {
...
+                transcript_selection_mode                           = transcript_selection_mode,
+                transcript_selection_list                           = transcript_selection_list,
+                annotation_defaults                                 = annotation_defaults,
+                annotation_overrides                                = annotation_overrides,
+                filter_funcotations                                 = filter_funcotations_or_default,
+#                interval_list
+                compress                                            = compress,
+#
+                allow_hg19_gencode_b37_contig_matching              = true,
+#                allow_hg19_gencode_b37_contig_matching_force_unsafe = false,
```
Comment (davidbenjamin) [inline scripts/mutect2_wdl/mutect2.wdl]: Delete?

File: scripts/mutect2_wdl/mutect2.wdl [hunk 4]
```diff
@@ -350,22 +350,33 @@ workflow Mutect2 {
...
+#                lookahead_cache_bp                                  = 100000,
```
Comment (davidbenjamin) [inline scripts/mutect2_wdl/mutect2.wdl]: Delete?

File: scripts/mutect2_wdl/mutect2.wdl [hunk 5]
```diff
@@ -976,34 +987,55 @@ task SumFloats {
...
+    # =======================================
+    # Optional Inputs:
     File? data_sources_tar_gz
     String? transcript_selection_mode
     Array[String]? transcript_selection_list
     Array[String]? annotation_defaults
     Array[String]? annotation_overrides
-    Boolean filter_funcotations
+    Boolean? filter_funcotations
+    File? interval_list
```
Comment (davidbenjamin) [inline scripts/mutect2_wdl/mutect2.wdl]: Can you call this `intervals` for consistency?

File: scripts/mutect2_wdl/mutect2.wdl [hunk 6]
```diff
@@ -1045,28 +1081,128 @@ task Funcotate {
         gatk --java-options "-Xmx${command_mem}m" Funcotator \
             --data-sources-path $DATA_SOURCES_FOLDER \
             --ref-version ${reference_version} \
+            --output-file-format ${output_format} \
             -R ${ref_fasta} \
-            -V ${m2_vcf} \
-            -O ${output_vcf} \
+            -V ${input_vcf} \
+            -O ${output_file} \
+            ${interval_list_arg} ${default="" interval_list} \
```
Comment (davidbenjamin) [inline scripts/mutect2_wdl/mutect2.wdl]: Replace this line with `${"-L " + interval_list} \`

File: scripts/mutect2_wdl/mutect2.wdl [hunk 7]
```diff
@@ -1045,28 +1081,128 @@ task Funcotate {
...
     output {
-        File funcotated_vcf = "${output_vcf}"
-        File funcotated_vcf_index = "${output_vcf_index}"
+        File funcotated_output = "${output_file}"
+        File funcotated_index  = "${output_index}"
     }
-}
+ }
+
+#task Funcotate {
```
Comment (davidbenjamin) [inline scripts/mutect2_wdl/mutect2.wdl]: Delete?

**Comment by LeeTL1220** [comment]
@jonn-smith @davidbenjamin Closing this PR, unmerged.  Jonn and I believe that the content here has already been merged or is no longer important.

**Pull request closed without merging**